src/App.js

```diff
@@ -2,30 +2,49 @@ import React, { useEffect, useState } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem("savedTodoList")) || []
-  );
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-};
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem("savedTodoList")),
```

Comment:
There's one small bug that got reintroduced when you changed how you're retrieving data from `localStorage`. Try deleting the `localStorage` data from your Application tab here: 

<img width="960" alt="Screen Shot 2023-08-09 at 8 02 20 PM" src="https://github.com/evaw277/react-todo/assets/12771651/35b064fb-caf2-4ee7-a57a-02eeafbac6a5">

Then if you refresh the page, you should get a crash. You had a solution for this in your existing code. Can you see what might have changed about your data retrieval in this PR?